CMake/folly-deps.cmake

```diff
@@ -197,7 +197,7 @@ if(NOT FOLLY_CPP_ATOMIC_BUILTIN)
   check_cxx_source_compiles("
     #include <atomic>
     int main(int argc, char** argv) {
-      struct Test { int val; };
+      struct Test { bool val; };
```

Comment:
What are the reason for these pair of changes? This should still be correct with your change, just want to make sure the change actually makes sense.

Reply:
For 8 and 16 bits types, on older versions of gcc, it would rely on `libatomic` to be linked to implement the atomic operations because the stream of instructions needed is too long to be inlined. That would not trigger for 32 and 64 bits types as there are a short sequence of instructions available in RISC-V for such instructions that can then be inlined.

The change here is to make sure we indeed hit that sub-word case.